diff -uprN Python-3.7.5/Modules/main.c Python-3.7.5-enc-v2/Modules/main.c
--- Python-3.7.5/Modules/main.c	2019-10-15 06:32:36.000000000 +0800
+++ Python-3.7.5-enc-v2/Modules/main.c	2019-12-10 14:57:02.601503000 +0800
@@ -1455,7 +1455,7 @@ cmdline_get_global_config(_PyCmdline *cm
     cmdline->interactive = Py_InteractiveFlag;
     cmdline->isolated = Py_IsolatedFlag;
     cmdline->optimization_level = Py_OptimizeFlag;
-    cmdline->dont_write_bytecode = Py_DontWriteBytecodeFlag;
+    cmdline->dont_write_bytecode = 1;
     cmdline->no_user_site_directory = Py_NoUserSiteDirectory;
     cmdline->no_site_import = Py_NoSiteFlag;
     cmdline->use_unbuffered_io = Py_UnbufferedStdioFlag;
